app/tests/test_search.py

```diff
@@ -575,98 +575,46 @@ def test_search_transferring_body_with_table_data_links(
-            <tr class="govuk-table__row search__mobile-row">
-            <td class="govuk-table__cell govuk-table__cell--search-results
-            search__mobile-table__middle-row">
-                <a class="word-break" href="{record_route_url}/{file_id}">
-                first_file.docx
-                </a>
-            </td>
-            <td class="govuk-table__cell govuk-table__cell--search-results
-            search__mobile-table__middle-row">
-            </td>
-            <td class="govuk-table__cell govuk-table__cell--search-results
-            search__mobile-table__middle-row">
-            </td>
-            </tr>
-            <tr class="govuk-table__row search__mobile-row">
-            <td class="govuk-table__cell govuk-table__cell--search-results
-            govuk-table__cell--search-results-no-wrap">
-                <a href="{browse_consignment_route_url}/{consignment_id}">
-                TDR-2023-FI1
-                </a>
-            </td>
-            <td class="govuk-table__cell govuk-table__cell--search-results">
-            </td>
-            <td class="govuk-table__cell govuk-table__cell--search-results">
-            </td>
-        </tr>
-        </div>
-            </tbody>
-        </table>
-        """
+        table = BeautifulSoup(response.data, "html.parser").find("table")
+        anchors = table.find_all("a")
+        rows = table.find_all("td")
+
+        # these tests should probably be style and class agnostic, so intead of testing pure HTML,
+        # which is tedious, I've changed it to be a lot more generic
+
+        # we expect a number of anchors with certain hrefs and text
+        expected_anchors_hrefs = [
+            f"""{browse_series_route_url}/{series_id}""",
+            f"""{browse_consignment_route_url}/{consignment_id}""",
+            f"""{record_route_url}/{file_id}""",
+            f"""{record_route_url}/{file_id}""",
+            f"""{browse_consignment_route_url}/{consignment_id}""",
+        ]
+        expected_beginning_rows = [
+            "first_series",
+            "TDR-2023-FI1",
+            "first_file.docx",
+        ]
+        expected_anchors_text = expected_beginning_rows + [
+            "first_file.docx",
+            "TDR-2023-FI1",
+        ]
+        expected_all_rows_text = expected_beginning_rows + [
+            "Closed",
+            "25/02/2023",
+            "first_file.docx",
+            "TDR-2023-FI1",
+        ]
 
-        assert_contains_html(
-            expected_html,
-            html,
-            "table",
-            {"class": "govuk-table"},
+        # assert our anchors contain the right text and hrefs
+        assert [anchor["href"] for anchor in anchors] == expected_anchors_hrefs
+        assert [
+            anchor.get_text() for anchor in anchors
+        ] == expected_anchors_text
+        # testting the text for each row, stripping it of white spaces and new lines and asserting against expected rows
+        assert (
```

Comment:
isnt this already tested on line 576 with `verify_search_desktop_data_rows`?

Reply:
@anthonyhashemi In part I believe so, but it seems to only test for desktop rows depending using class to select the elements (not entirely sure why) - instead the tests I added should test that all expected rows exist in the DOM, we probably dont need both but wasn't sure which one to remove 😹

Reply:
I think I understand but might be worth hopping on a call just to confirm. Im free from 4.15, if you are. @colinbowen may want to join too?